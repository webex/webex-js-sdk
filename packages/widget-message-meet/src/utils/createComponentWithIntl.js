import React from 'react';
import renderer from 'react-test-renderer';
import {IntlProvider} from 'react-intl';

function createComponentWithIntl(children, props = {locale: `en`}) {
  return renderer.create(
    <IntlProvider {...props}>
      {children}
    </IntlProvider>
  );
}

export default createComponentWithIntl;
