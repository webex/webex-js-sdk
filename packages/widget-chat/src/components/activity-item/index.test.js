import React from 'react';

import createComponentWithIntl from '../../utils/createComponentWithIntl';

import ActivityItem from '.';

describe(`ActivityItem post component is self`, () => {
  const activity = {
    id: `test-123-123-123-123`,
    isSelf: true,
    activity: {
      displayName: `Test Activity Content`
    },
    onActivityDelete: jest.fn(),
    name: `Test User`,
    timestamp: `2016-09-20T19:52:57.186Z`,
    verb: `post`
  };

  const component = createComponentWithIntl(
    <ActivityItem {...activity} />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});

describe(`ActivityItem post component not self`, () => {
  const activity = {
    id: `test-123-123-123-123`,
    isSelf: false,
    activity: {
      displayName: `Test Activity Content`
    },
    name: `Test User`,
    timestamp: `2016-09-20T19:52:57.186Z`,
    verb: `post`
  };

  const component = createComponentWithIntl(
    <ActivityItem {...activity} />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});

describe(`ActivityItem post component not self with html`, () => {
  const activity = {
    id: `test-123-123-123-123`,
    isSelf: false,
    activity: {
      content: `<pre>js.func();</pre>`,
      displayName: `Test Activity Content`
    },
    name: `Test User`,
    timestamp: `2016-09-20T19:52:57.186Z`,
    verb: `post`
  };

  const component = createComponentWithIntl(
    <ActivityItem {...activity} />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});


describe(`ActivityItem tombstone component`, () => {
  const activity = {
    id: `test-123-123-123-123`,
    activity: {
      displayName: `Test Activity Content`
    },
    name: `Test User`,
    timestamp: `2016-09-20T19:52:57.186Z`,
    verb: `tombstone`
  };

  const component = createComponentWithIntl(
    <ActivityItem {...activity} />
  );

  it(`renders properly`, () => {
    expect(component).toMatchSnapshot();
  });
});
