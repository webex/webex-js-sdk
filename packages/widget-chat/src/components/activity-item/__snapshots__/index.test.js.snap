exports[`ActivityItem post component is self renders properly 1`] = `
<div
  className="activity-item-container">
  <div
    className="activity-item">
    <div
      className="avatar-wrapper">
      <div
        className="avatar"
        style={undefined}>
        <span
          className="avatar-self" />
      </div>
    </div>
    <div
      className="content-container">
      <div
        className="meta">
        <div
          className="display-name"
          title="{name}">
          Test User
        </div>
        <div
          className="published">
          2016-09-20T19:52:57.186Z
        </div>
      </div>
      <div
        className="activity-content">
        <div
          className="activity-text">
          Test Activity Content
        </div>
      </div>
    </div>
    <div
      className="activity-post-actions">
      <div
        className="activity-post-action">
        <div
          className="post-action-item">
          <div
            className="icon-button"
            onClick={[Function handleOnFlag]}>
            <div
              className="icon"
              title="Flag this message" />
          </div>
        </div>
      </div>
      <div
        className="activity-post-action">
        <div
          className="post-action-item">
          <div
            className="icon-button"
            onClick={[Function handleOnDelete]}>
            <div
              className="icon"
              title="Delete this message" />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ActivityItem post component not self renders properly 1`] = `
<div
  className="activity-item-container">
  <div
    className="activity-item">
    <div
      className="avatar-wrapper">
      <div
        className="avatar"
        style={undefined}>
        <span
          className="avatar-letter">
          T
        </span>
      </div>
    </div>
    <div
      className="content-container">
      <div
        className="meta">
        <div
          className="display-name"
          title="{name}">
          Test User
        </div>
        <div
          className="published">
          2016-09-20T19:52:57.186Z
        </div>
      </div>
      <div
        className="activity-content">
        <div
          className="activity-text">
          Test Activity Content
        </div>
      </div>
    </div>
    <div
      className="activity-post-actions">
      <div
        className="activity-post-action">
        <div
          className="post-action-item">
          <div
            className="icon-button"
            onClick={[Function handleOnFlag]}>
            <div
              className="icon"
              title="Flag this message" />
          </div>
        </div>
      </div>
      <div
        className="activity-post-action">
        <div
          className="action-spacer" />
      </div>
    </div>
  </div>
</div>
`;

exports[`ActivityItem tombstone component renders properly 1`] = `
<div
  className="activity-item-container">
  <div
    className="system-message">
    Test User deleted their own message. 2016-09-20T19:52:57.186Z
  </div>
</div>
`;
